### PR TITLE
Venue selector

### PR DESCRIPTION
**Pull request by Androxium**
Added the sliding drawer to the way-finding sample app that allows users to change the venue being displayed.

### CODE REVIEW

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java [hunk 1]
```diff
@@ -204,10 +219,140 @@ public void onPause() {
...
+        drawer.setDrawerListener(toggle);
+        toggle.syncState();
+
         context = this;
         self = this;
-        setContentView(R.layout.activity_main);
+
         AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
+
+        LocationGenerator amenity = new LocationGenerator() {
```
Comment (cbmaiolo) [inline wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: All this location generator stuff shouldn't be here. We should move it to the VenuesCallback.

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java [hunk 2]
```diff
@@ -204,10 +219,140 @@ public void onPause() {
...
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new CustomerLocation(data, _index, venue);
+            }
+        };
+        final LocationGenerator[] locationGenerators2 = {customerLocation};
+
+        // Map View
+        mapView = (MapView) getSupportFragmentManager().findFragmentById(R.id.map_fragment);
+        mapView.setDelegate(this);
+        mapComparator = new Comparator<Map>() {
```
Comment (cbmaiolo) [inline wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: Any reason we moved this out of the callback?

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java [hunk 3]
```diff
@@ -491,36 +632,6 @@ public void onClick(DialogInterface dialog, int which) {
...
-                }
-            }
-
-            @Override
-            public void onError(Exception exception) {
-
-            }
-        };
-
         MappedinCallback<List<Venue>> getVenuesCallback = new VenuesCallback(mappedIn, getVenueCallback);
```
Comment (cbmaiolo) [inline wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: maybe we should make this getVenuesCallback reusable so we can use it in your onItemClicked

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java [hunk 4]
```diff
@@ -204,10 +219,140 @@ public void onPause() {
...
+                venueListAdapter.setVenues(venues);
+                venueList.setAdapter(venueListAdapter);
+                venueList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(final AdapterView<?> parent, View view, final int position, long id) {
+                        venueProgressBar.setVisibility(View.VISIBLE);
+                        mapView.getView().setVisibility(View.INVISIBLE);
+                        levelPickerLayout.setVisibility(View.INVISIBLE);
+                        activeVenue = venues.get(position);
+                        mappedIn.getVenue(activeVenue, locationGenerators2, getVenueCallback);
```
Comment (cbmaiolo) [inline wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: we shouldn't be using locationGenerators2 here, we should use the same getVenues as when we first launch the app

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java (hunk 2 above)
Comment (Androxium) [reply wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: I was trying to improve memory usage by reducing the amount of new objects in functions we called multiple times.

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java (hunk 4 above)
Comment (Androxium) [reply wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: I'm not quite following what you mean. The _getVenues_ function used to initially start the app will automatically default to the first venue the user has access to (lines 256-288 for reference). _getVenue_ will get the full details of a specified venue. Following the Android Docs, I call _getVenue_ when the user wishes to change venues. As one of the parameters for _getVenue_ is locationGenerators2, that is why I pass it. If you try to override the default venue with activeVenue, you run into an _IndexOutOfBoundsException_.

Comment (cbmaiolo) [reply wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java]: Gotcha

File: wayfindingsample/src/main/java/mappedin/com/wayfindingsample/VenuesCallback.java [hunk 5]
```diff
@@ -16,9 +16,14 @@
 public class VenuesCallback implements MappedinCallback<List<Venue>> {
 
     private MappedIn mappedIn;
-    private LocationGenerator[] customerLocations;
     private MappedinCallback<Venue> getVenueCallback;
 
+    private LocationGenerator[] customerLocations;
```
Comment (cbmaiolo) [inline wayfindingsample/src/main/java/mappedin/com/wayfindingsample/VenuesCallback.java]: Lets use better names to explain what these are:

// Use this location generator if your data schema has only one location type with externalIds
// Most keys use this data schema
private LocationGenerator[]  genericLocations;

// Use this generator if your data schema uses multiple complex Location types 
// (tenant, amenity, elevator, escalatorStairs) you might have a different data schema
// A mappedin representative can help you set up any missing types
private LocationGenerator[] detailedLocations;

**Pull request merged**